appinventor/components/src/com/google/appinventor/components/runtime/ChartDataBase.java

```diff
@@ -1,62 +1,111 @@
 package com.google.appinventor.components.runtime;
 
 import com.github.mikephil.charting.charts.Chart;
-import com.github.mikephil.charting.data.ChartData;
+import com.google.appinventor.components.annotations.DesignerProperty;
+import com.google.appinventor.components.annotations.PropertyCategory;
+import com.google.appinventor.components.annotations.SimpleObject;
+import com.google.appinventor.components.annotations.SimpleProperty;
+import com.google.appinventor.components.common.PropertyTypeConstants;
+import com.google.appinventor.components.runtime.util.ChartModel;
+import com.google.appinventor.components.runtime.util.ElementsUtil;
 
-import java.util.HashSet;
+@SimpleObject
+public abstract class ChartDataBase implements Component {
+    protected ChartBase container = null;
+    protected ChartModel chartModel = null;
 
-public abstract class ChartDataBase<T extends ChartData> extends AndroidNonvisibleComponent {
+    private String label;
+    private int color;
 
-    protected T chartData;
-    protected HashSet<ChartBase> charts;
+    /**
+     * Creates a new Chart Data component.
+     */
+    public ChartDataBase(ChartBase chartContainer) {
+        this.container = chartContainer;
+
+        chartModel = chartContainer.createChartModel();
+
+        // Set default values
+        Color(Component.COLOR_BLACK);
+        Label("");
+    }
 
     /**
-     * Creates a new ChartDataBase component.
+     * Returns the data series color as an alpha-red-green-blue integer.
      *
-     * @param form the container that this component will be placed in
+     * @return  background RGB color with alpha
      */
-    protected ChartDataBase(Form form) {
-        super(form);
-
-        charts = new HashSet<ChartBase>();
+    @SimpleProperty(
+            category = PropertyCategory.APPEARANCE)
+    public int Color() {
+        return color;
     }
 
     /**
-     * Getter method for the underlying data object of the ChartData component
+     * Specifies the data series color as an alpha-red-green-blue integer.
      *
-     * @return - ChartData object
+     * @param argb  background RGB color with alpha
      */
-    public T getChartData() {
-        return chartData;
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_COLOR,
+            defaultValue = Component.DEFAULT_VALUE_COLOR_BLACK)
+    @SimpleProperty
+    public void Color(int argb) {
+        color = argb;
+        chartModel.setColor(color);
+        refreshChart();
```

Comment:
Every property setter calls refreshChart(). Any way to reduce redundancy? Maybe an event method that I can overwrite?